src/task.c

```diff
@@ -721,26 +782,38 @@ void jl_init_root_task(void *stack, size_t ssize)
     ptls->current_task->ssize = ssize;
     ptls->current_task->stkbuf = stack;
 #endif
-    ptls->current_task->started = 1;
-    ptls->current_task->parent = ptls->current_task;
-    ptls->current_task->current_module = ptls->current_module;
+#ifdef JULIA_ENABLE_PARTR
+    ptls->current_task->settings = TASK_IS_STICKY | TASK_IS_DETACHED;
+    ptls->current_task->next = NULL;
+    ptls->current_task->storage = jl_nothing;
+    ptls->current_task->current_tid = ptls->tid;
+    ptls->current_task->sticky_tid = ptls->tid;
+    ptls->current_task->parent = NULL;
+    ptls->current_task->arr = NULL;
+    ptls->current_task->red = NULL;
+    ptls->current_task->red_result = jl_nothing;
+    ptls->current_task->grain_num = -1;
```

Comment:
This is missing some field initializations: `args`, `mfunc`, `rargs`, `mredfunc`, etc. The gc is crashing when it hits uninitialized pointers here.

Reply:
A `jl_value_t` array like `args` (or any of the `jl_any_type` fields) should be `jl_nothing` rather than NULL, yes?

Reply:
Generally yes. If set to NULL, a field will be "undefined" in julia and raise an error on access. In that case the main worry is the `ninitialized` argument to `jl_new_datatype`.